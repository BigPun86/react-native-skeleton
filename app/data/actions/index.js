/**
 * Created by maluramichael on 18/08/16.
 */

import * as User from './user';
import * as Loading from './loading';
import * as SideMenu from './sideMenu';

export { User, Loading, SideMenu };