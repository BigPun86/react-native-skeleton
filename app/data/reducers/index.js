/**
 * Created by maluramichael on 18/08/16.
 */

import { combineReducers } from 'redux';
import Loading from './loading';

export default combineReducers( {
	Loading
} );