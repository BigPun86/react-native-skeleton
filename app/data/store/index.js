/**
 * Created by maluramichael on 18/08/16.
 */
import Reactotron from 'reactotron-react-native';
import createReactotronEnhancer from 'reactotron-redux';
import {
	applyMiddleware,
	createStore,
	compose
} from 'redux';
import Thunk from 'redux-thunk';
import CombinedReducers from '../reducers';

/*
 MIDDLEWARE (https://github.com/reactjs/redux/blob/master/docs/api/applyMiddleware.md)
 Middleware is the suggested way to extend Redux with custom functionality. Middleware lets you wrap the store's
 dispatch method for fun and profit. The key feature of middleware is that it is composable. Multiple middleware can
 be combined together, where each middleware requires no knowledge of what comes before or after it in the chain.
 */

/*
 COMPOSE (https://github.com/reactjs/redux/blob/master/docs/api/compose.md)

 Composes functions from right to left.
 This is a functional programming utility, and is included in Redux as a convenience.
 You might want to use it to apply several store enhancers in a row.

 (arguments): The functions to compose. Each function is expected to accept a single parameter. Its return value will be provided as an argument to the function standing to the left, and so on. The exception is the right-most argument which can accept multiple parameters, as it will provide the signature for the resulting composed function.
 returns (Function): The final function obtained by composing the given functions from right to left.
 */

/*
 STORE ENHANCER (https://github.com/reactjs/redux/blob/master/docs/Glossary.md#store-enhancer)
 A store enhancer is a higher-order function that composes a store creator to return a new, enhanced store creator.
 This is similar to middleware in that it allows you to alter the store interface in a composable way.

 Store enhancers are much the same concept as higher-order components in React, which are also occasionally called
 “component enhancers”.

 Because a store is not an instance, but rather a plain-object collection of functions, copies can be easily
 created and modified without mutating the original store. There is an example in compose documentation
 demonstrating that.

 Most likely you'll never write a store enhancer, but you may use the one provided by the developer tools. It is
 what makes time travel possible without the app being aware it is happening. Amusingly, the Redux middleware
 implementation is itself a store enhancer.
 */
const enhancerReactNativeDebugger = global.reduxNativeDevTools ? global.reduxNativeDevTools() : nope => nope;
const enhancerReactotron = createReactotronEnhancer( Reactotron );

const middleware = applyMiddleware( Thunk );

const composeEnhancers = compose( enhancerReactotron, middleware, enhancerReactNativeDebugger );
const storeEnhancer = composeEnhancers( createStore );

const store = storeEnhancer( CombinedReducers );
export default store;