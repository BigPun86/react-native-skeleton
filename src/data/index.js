/**
 * Created by maluramichael on 18/08/16.
 */

import * as Actions from './actions';
import * as Reducers from './reducers';
import Store from './store';

export {
	Actions,
	Store,
	Reducers
}