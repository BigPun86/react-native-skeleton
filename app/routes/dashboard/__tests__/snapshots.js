/**
 * Created by maluramichael on 05/10/16.
 */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Navigator from 'react-native-better-navigator';
import Component from '../';
import {Store} from '../../../data';

it('renders correctly without props', () => {
	const tree = renderer.create(
		<Provider store={Store}>
			<Navigator initialRoute={{name: 'Route'}} routes={()=>{return Component}}/>
		</Provider>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
