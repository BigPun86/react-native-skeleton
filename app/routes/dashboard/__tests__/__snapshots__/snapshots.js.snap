exports[`test renders correctly without props 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }>
    <View
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "marginTop": 64,
            },
            Object {},
            undefined,
          ],
          null,
        ]
      }>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
          }
        } />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
        },
        Array [
          Object {
            "backgroundColor": "#333",
          },
          Object {
            "backgroundColor": "#777",
          },
        ],
      ]
    }>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": 44,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Array [
              undefined,
            ],
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              Array [
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                },
                undefined,
              ],
            ]
          }>
          
        </Text>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 44,
          "left": 0,
          "opacity": 1,
          "overflow": "hidden",
          "position": "absolute",
          "top": 20,
        }
      }>
      <View />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "transparent",
          "height": 44,
          "opacity": 1,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }>
      <View />
    </View>
  </View>
</View>
`;
